### ISSUE TITLE
Mark all as read for a feed (feature request)

### ISSUE THREAD

**Issue by GLLM** [SSilence/selfoss#279]
Hi,

I have some feeds that produce more than 400 articles a day (news).
Some day, I can read some of them (articles) and mark the other as read as the day goes by.
On some other days, I cannot.

The idea would be to have a "Mark all as read" that would mark all feed's items as read, and not just the one loaded in the browser.
Indeed, with 100 articles loaded at a time, I sometime need to press "Mark as read" almost 10 times.

(This would not do the same as mark as read since it would mark all items from a feed as read in the DB, and not only those loaded in the browser)

Thanks,
GLLM

**Comment by jantman** [comment]
This seems to exist in 2.17.

**Comment by jtojnar** [comment]
@jantman Unfortunately, the button only marks the visible items as read. This feature request wants to mark every single one.

**Comment by jantman** [comment]
@jtojnar  Ah ok, sorry, my mistake. Most of the feeds I follow provide fewer items than my pagination setting.

**Comment by xsteadfastx** [comment]
i would love a mark all as read button and not just the visible ones.

**Comment by perryflynn** [comment]
+1

**Comment by lapete** [comment]
+1 this feature would really help. Just marked 4600 entries as read because I didn't check my feed for a few days, I would appreciate a "mark all read" too.

**Comment by jtojnar** [comment]
@PhrozenByte 

> What's the sense about marking only visible/loaded entries anyway?

Yeah, I do not think it makes much sense when infinite scrolling (`auto_stream_more`) is enabled. Changing the button to really only mark the items visible on the screen (e.g. using [intersection observers](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)) would be fine with me.

@girvo’s idea to add a submenu sounds reasonable too.
